internal/buffer/buffer.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2013 Google Inc. All Rights Reserved.
```

Comment:
@dims : I just moved the code, so keeping the original copyright seemed appropriate. Should I also add "Copyright 2022 The Kubernetes contributors"?

Reply:
yes please add